magenta-lib/src/test/scala/magenta/input/resolver/DeploymentTasksGraphBuilderTest.scala

```diff
@@ -0,0 +1,62 @@
+package magenta.input.resolver
+
+import magenta.graph.{EndNode, Graph, MidNode, StartNode}
+import magenta.input.Deployment
+import org.scalatest.{FlatSpec, ShouldMatchers}
+
+class DeploymentTasksGraphBuilderTest extends FlatSpec with ShouldMatchers {
```

Comment:
I think this needs a test for when there is a long chain of dependencies (see my comment above about the implementation of the graph builder).